### PR TITLE
Add steel ladder

### PR DESCRIPTION
**Pull request by kilbith**
See discussion : https://github.com/minetest/minetest_game/issues/883

Requires 7 steel ingots for crafting, yields 15 ladders since the material is fairly expensive.

a

### CODE REVIEW

**Comment by 0-afflatus** [comment]
LGTM

File: mods/default/nodes.lua [hunk 1]
```diff
@@ -1676,6 +1677,28 @@ minetest.register_node("default:ladder", {
...
+	walkable = false,
+	climbable = true,
+	is_ground_content = false,
+	selection_box = {
+		type = "wallmounted",
+		--wall_top = = <default>
+		--wall_bottom = = <default>
+		--wall_side = = <default>
+	},
+	groups = {cracky = 2, oddly_breakable_by_hand = 3},
```
Comment (PilzAdam) [inline mods/default/nodes.lua]: Don't make this `oddly_breakable_by_hand`.

File: mods/default/nodes.lua [hunk 2]
```diff
@@ -1676,6 +1677,28 @@ minetest.register_node("default:ladder", {
...
+	legacy_wallmounted = true
```
Comment (PilzAdam) [inline mods/default/nodes.lua]: This is only needed for nodes that used wallmounted in 0.3

**Comment by PilzAdam** [comment]
It seems like the nodedef is missing `sound`.

Maybe rename the old wooden ladder to `default:ladder_wood`?

Btw, the recipe for the wooden ladder should give more ladders, too.

**Comment by kilbith** [comment]
@PilzAdam Addressed your line-notes.

> It seems like the nodedef is missing sounds.

Not required since it's defaulting to `default.node_sound_defaults()` if `sounds` ain't specified. But I added it anyway.

> Maybe rename the old wooden ladder to default:ladder_wood?

Done.

> the recipe for the wooden ladder should give more ladders

How many it should have for you ?

**Comment by PilzAdam** [comment]
You forgot to update the output of the crafting recipe for `default:ladder`. If you update this, then :+1:.

> How many it should have for you ?

Dunno, something about 2 to 5? This is a different issue, though.

**Comment by kilbith** [comment]
OK, updated. Yield for the wooden ladder set to 3 but need feedbacks from other devs.

**Comment by MT-Modder** [comment]
:+1:

**Comment by Wuzzy2** [comment]
>> Maybe rename the old wooden ladder to default:ladder_wood?

>Done.

Congratulations, you just broke all worlds which have `default:ladder` in it. xD
Currently, there is no alias for `default:ladder` in the commit.

IMO renaming itemstrings should be avoided, if possible. The user doesn't care about the itemstring. What's more important is the user-visible description, but this has been changed already.

**Comment by kilbith** [comment]
> Currently, there is no alias for default:ladder in the commit.

... what's [that](https://github.com/minetest/minetest_game/pull/892/files#diff-475164699c029569dca6e8401b7f6d62R78) ?

**Comment by Thomas--S** [comment]
:+1:

**Comment by paramat** [comment]
Iron bars from xpanes is snappy=2, cracky=3 and steel doors have snappy too, so perhaps add snappy=2 to this?

**Comment by kilbith** [comment]
Last time I added `snappy = 2` it had no effect. But OK - added.

**Comment by paramat** [comment]
:+1:

**Comment by PilzAdam** [comment]
Wait what. Steel isn't "snappy" at all. Please remove that.

**Comment by kilbith** [comment]
@PilzAdam Done. Also removed from xpanes bars.

**Comment by Gael-de-Sailly** [comment]
:+1:

**Comment by paramat** [comment]
Heh ok. steel doors are snappy too but that can be changed later, will this merge soon.

**Comment by paramat** [comment]
7f317871d48bd261faa16bbe2cfc87e551d2ae6e

**Pull request closed without merging**